File: SDKs/yodlee.com/1.1.0/typescript/src/sdk/models/operations/updatetransactioncategorizationrule.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class UpdateTransactionCategorizationRulePathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=ruleId" })
  ruleId: number;
}


export class UpdateTransactionCategorizationRuleRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: UpdateTransactionCategorizationRulePathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request: shared.TransactionCategorizationRuleRequest;
}


export class UpdateTransactionCategorizationRuleResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  yodleeError?: shared.YodleeError;
}

File: SDKs/yodlee.com/1.1.0/python/src/sdk/models/operations/updatetransactioncategorizationrule.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from ..shared import transactioncategorizationrulerequest as shared_transactioncategorizationrulerequest
from ..shared import yodleeerror as shared_yodleeerror


@dataclasses.dataclass
class UpdateTransactionCategorizationRulePathParams:
    rule_id: int = dataclasses.field(metadata={'path_param': { 'field_name': 'ruleId', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class UpdateTransactionCategorizationRuleRequest:
    path_params: UpdateTransactionCategorizationRulePathParams = dataclasses.field()
    request: shared_transactioncategorizationrulerequest.TransactionCategorizationRuleRequest = dataclasses.field(metadata={'request': { 'media_type': 'application/json' }})
    

@dataclasses.dataclass
class UpdateTransactionCategorizationRuleResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    yodlee_error: Optional[shared_yodleeerror.YodleeError] = dataclasses.field(default=None)
    
